src/apps/beanstalk/ethereum/beanstalk.silo.contract-position-fetcher.ts

```diff
@@ -0,0 +1,92 @@
+import { Inject } from '@nestjs/common';
+import { BigNumber, BigNumberish, ethers } from 'ethers';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { PositionTemplate } from '~app-toolkit/decorators/position-template.decorator';
+import { gqlFetch } from '~app-toolkit/helpers/the-graph.helper';
+import { DefaultDataProps } from '~position/display.interface';
+import { MetaType } from '~position/position.interface';
+import { ContractPositionTemplatePositionFetcher } from '~position/template/contract-position.template.position-fetcher';
+import {
+  GetDefinitionsParams,
+  DefaultContractPositionDefinition,
+  GetTokenDefinitionsParams,
+  UnderlyingTokenDefinition,
+  GetDisplayPropsParams,
+  GetTokenBalancesParams,
+} from '~position/template/contract-position.template.types';
+
+import { BEANSTALK_ADDRESS, Token, GRAPH_URL, silos } from '../common/beanstalk.helper';
+import { Beanstalk, BeanstalkContractFactory } from '../contracts';
+import { GET_DEPOSIT } from '../graphql/getSiloDeposit';
+
+export type SiloDefinition = {
+  address: string;
+  underlying: Token[];
+  siloName: string;
+};
+
+@PositionTemplate()
+export class EthereumBeanstalkSiloContractPositionFetcher extends ContractPositionTemplatePositionFetcher<Beanstalk> {
+  groupLabel = 'Silo Deposits';
+
+  constructor(
+    @Inject(APP_TOOLKIT) protected readonly appToolkit: IAppToolkit,
+    @Inject(BeanstalkContractFactory) protected readonly beanstalkContractFactory: BeanstalkContractFactory,
+  ) {
+    super(appToolkit);
+  }
+
+  getContract(_address: string): Beanstalk {
+    return this.beanstalkContractFactory.beanstalk({ address: BEANSTALK_ADDRESS, network: this.network });
+  }
+
+  getDefinitions(_params: GetDefinitionsParams): Promise<SiloDefinition[]> {
+    return Promise.resolve(
+      Object.keys(silos).map(address => ({
+        address: address,
+        underlying: silos[address].underlying,
+        siloName: silos[address].name,
+      })),
+    );
+  }
+
+  getTokenDefinitions({
+    definition,
+  }: GetTokenDefinitionsParams<Beanstalk, SiloDefinition>): Promise<UnderlyingTokenDefinition[] | null> {
+    return Promise.resolve(
+      definition.underlying.map(bt => ({
+        metaType: MetaType.SUPPLIED,
+        address: bt.address,
+        network: this.network,
+      })),
+    );
+  }
+
+  getLabel({
```

Comment:
adding `async` in front of getLabel will enable you not to have to wrap everything in Promise.resolve()